cli/cmd/add/component-version/cmd.go

```diff
@@ -0,0 +1,359 @@
+	ComponentVersionConflictPolicyReplace      ComponentVersionConflictPolicy = "replace"
+)
+
+func (p ComponentVersionConflictPolicy) ToConstructorConflictPolicy() constructor.ComponentVersionConflictPolicy {
+	switch p {
+	case ComponentVersionConflictPolicyReplace:
+		return constructor.ComponentVersionConflictReplace
+	case ComponentVersionConflictPolicySkip:
+		return constructor.ComponentVersionConflictSkip
+	default:
+		return constructor.ComponentVersionConflictAbortAndFail
+	}
+}
+
+func ComponentVersionOverridePolicies() []string {
+	return []string{
+		string(ComponentVersionConflictPolicyAbortAndFail),
+		string(ComponentVersionConflictPolicySkip),
+		string(ComponentVersionConflictPolicyReplace),
+	}
+}
+
+func New() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:        "component-version",
+		Aliases:    []string{"cv", "component-versions", "cvs"},
+		SuggestFor: []string{"component", "components", "version", "versions"},
+		Short:      fmt.Sprintf("Add component version(s) to an OCM Repository stored as Common Transport Format Archive (CTF) based on a %[1]q file", DefaultComponentConstructorBaseName),
+		Args:       cobra.NoArgs,
+		Long: fmt.Sprintf(`Add component version(s) to an OCM Common Transport Format Archive (CTF) that can be reused
+for transfers.
+
+A %[1]q file is used to specify the component version(s) to be added. It can contain both a single component or many
+components. The component reference is used to determine the repository to add the components to.
+
+By default, the command will look for a file named "%[1]q.yaml" or "%[1]q.yml" in the current directory.
+If given a path to a directory, the command will look for a file named "%[1]s.yaml" or "%[1]s.yml" in that directory.
+If given a path to a file, the command will attempt to use that file as the %[1]q file.
+
+In case the component archive does not exist, it will be created by default.
+`,
+			DefaultComponentConstructorBaseName,
+		),
+		Example: strings.TrimSpace(fmt.Sprintf(`
+Adding component versions to a non-default CTF named %[2]q based on a non-default default %[4]q file:
+
+add component-version  --%[1]s ./path/to/%[2]s --%[3]s ./path/to/%[4]s.yaml
+`, FlagRepositoryRef, LegacyDefaultArchiveName, FlagComponentConstructorPath, DefaultComponentConstructorBaseName)),
+		RunE:              AddComponentVersion,
+		DisableAutoGenTag: true,
+	}
+
+	cmd.Flags().Int(FlagConcurrencyLimit, 4, "maximum amount of parallel requests to the repository for resolving component versions")
+	file.VarP(cmd.Flags(), FlagRepositoryRef, string(FlagRepositoryRef[0]), LegacyDefaultArchiveName, "path to the repository")
+	file.VarP(cmd.Flags(), FlagComponentConstructorPath, string(FlagComponentConstructorPath[0]), DefaultComponentConstructorBaseName+".yaml", "path to the repository")
+	cmd.Flags().String(FlagBlobCacheDirectory, filepath.Join(".ocm", "cache"), "path to the blob cache directory")
```

Comment:
Should this be on the path of the running command?

Reply:
Currently we dont have a central cache directory that we can configure via OCM config. I think the next best thing would be to introduce the old OCM v1 caching configuration. WDYT?

Reply:
Yah, works for me.